libraries/AP_GSOF/AP_GSOF.cpp

```diff
@@ -0,0 +1,248 @@
+#define ALLOW_DOUBLE_MATH_FUNCTIONS
+
+#include "AP_GSOF_config.h"
+#include "AP_GSOF.h"
+
+#if AP_GSOF_ENABLED
+
+
+
+#include <AP_Logger/AP_Logger.h>
+#include <AP_HAL/utility/sparse-endian.h>
+#include <GCS_MAVLink/GCS.h>
+#include <AP_GSOF/AP_GSOF.h>
+
+extern const AP_HAL::HAL& hal;
+
+#define gsof_DEBUGGING 0
+
+#if gsof_DEBUGGING
+# define Debug(fmt, args ...)                  \
+do {                                            \
+    hal.console->printf("%s:%d: " fmt "\n",     \
+                        __FUNCTION__, __LINE__, \
+                        ## args);               \
+    hal.scheduler->delay(1);                    \
+} while(0)
+#else
+# define Debug(fmt, args ...)
+#endif
+
+int
+AP_GSOF::parse(const uint8_t temp, const uint8_t n_expected)
+{
+    // https://receiverhelp.trimble.com/oem-gnss/index.html#API_DataCollectorFormatPacketStructure.html
+    switch (msg.state) {
+    default:
+    case Msg_Parser::State::STARTTX:
+        if (temp == STX) {
+            msg.state = Msg_Parser::State::STATUS;
+            msg.read = 0;
+            msg.checksumcalc = 0;
+        }
+        break;
+    case Msg_Parser::State::STATUS:
+        msg.status = temp;
+        msg.state = Msg_Parser::State::PACKETTYPE;
+        msg.checksumcalc += temp;
+        break;
+    case Msg_Parser::State::PACKETTYPE:
+        msg.packettype = temp;
+        msg.state = Msg_Parser::State::LENGTH;
+        msg.checksumcalc += temp;
+        break;
+    case Msg_Parser::State::LENGTH:
+        msg.length = temp;
+        msg.state = Msg_Parser::State::DATA;
+        msg.checksumcalc += temp;
+        break;
+    case Msg_Parser::State::DATA:
+        msg.data[msg.read] = temp;
+        msg.read++;
+        msg.checksumcalc += temp;
+        if (msg.read >= msg.length) {
+            msg.state = Msg_Parser::State::CHECKSUM;
+        }
+        break;
+    case Msg_Parser::State::CHECKSUM:
+        msg.checksum = temp;
+        msg.state = Msg_Parser::State::ENDTX;
+        if (msg.checksum == msg.checksumcalc) {
+            const auto n_processed = process_message();
+            if (n_processed != n_expected ) {
+                return UNEXPECTED_NUM_GSOF_PACKETS;
+            }
+            return n_processed;
+        }
+        break;
+    case Msg_Parser::State::ENDTX:
+        msg.endtx = temp;
+        msg.state = Msg_Parser::State::STARTTX;
+        break;
+    }
+
+    return NO_GSOF_DATA;
+}
+
+double
+AP_GSOF::SwapDouble(const uint8_t* src, const uint32_t pos) const
```

Comment:
I know it's existing code but we normally done use CamelCase... or maybe it's Pascal case.

Reply:
or remove the function